### DISCUSSION TITLE
Making the drag/spring(?) animation less heavy. Is this possible?

### DISCUSSION THREAD

**Discussion by DennisKraaijeveld** [davidjerleke/embla-carousel#789] [Help] [answered]
### Summary

Hi! So when dragging the items of the carousel to the end, or scrolling to the end of the carousel with your mouse, the animation or spring effect is so heavy and over exaggerated in my opinion. Would it be possible to adjust this and make it more subtle? 

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Hi @DennisKraaijeveld,

Thanks for your question. There's no convenient way of doing this but it's doable. You can **disable** [`ScrollBounds`](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/ScrollBounds.ts) like so:

```ts
emblaApi.internalEngine().scrollBounds.toggleActive(false); // Disable bounds
```

And after that you can write your own logic:

```tsx
emblaApi.on('scroll', (emblaApi) => {
  // Access the same utilities as ScrollBounds
  const {
    limit,
    offsetLocation,
    target,
    scrollBody,
    percentOfView,
    dragHandler,
  } = emblaApi.internalEngine();

  if (dragHandler.pointerDown()) { 
    // User has their pointer down
  } else {
    // No pointer down
  }
  
  // Write your own logic here...
});
```

You probably want to study [`ScrollBounds`](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/ScrollBounds.ts) if you want to do something similar. For example, the [`shouldConstrain`](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/ScrollBounds.ts#L24-L29) function will return `true` when the carousel has scrolled past its bounds.

Good luck!

Best,
David
